### ISSUE TITLE
Add an Option to Display Non-Constant Strings 

### ISSUE THREAD

**Issue by cztian** [GregoryMorse/GhidraDec#6]
Ghidra uses "Default settings" for string mutability, which Hex-Rays uses a decompiler flag for it. Is it feasible for Ghidra decompiler to display all string literals whether they are constant or not?
![image](https://user-images.githubusercontent.com/14105716/179746363-e5a76eee-8a04-4142-b0d2-9ae0b88ba62e.png)
![image](https://user-images.githubusercontent.com/14105716/179746759-8ea2003d-4b88-41b7-97ca-ad3e73b40047.png)

**Comment by GregoryMorse** [comment]
As far as I know this is being done by the plugin, the decompiler itself I dont believe contains such a setting.  The default string settings you show is part of the Ghidra Java frontend and nothing to do with the decompiler backend.  Which means this would basically be an additional feature for the plugin.  As far as I know it tries to display all strings, is this not happening, mutable strings arent displayed?

**Comment by cztian** [comment]
> 

Yes, strings in the mutable address parts are not displayed as strings, but as symbols.

**Comment by GregoryMorse** [comment]
GhidraDec already supports Ghidra's `getstringdata` callback and will provide string contents from IDA when the decompiler asks for a string item. The output prologue also prints string initializers for data that IDA has identified as strings, including writable data when IDA marks it as a string.

However, Ghidra's native decompiler does not appear to expose a Hex-Rays-style option to force all non-constant strings to display as literals. A broader "always annotate mutable strings" feature would need to be implemented in GhidraDec's viewer/prologue/comment layer rather than as a Ghidra decompiler option.

Closing this as not directly actionable as a decompiler option. We can track viewer-side mutable string annotations separately if needed.
